src/test/java/seedu/pill/command/HelpCommandTest.java

```diff
@@ -0,0 +1,168 @@
+        assertTrue(output.contains("add:"));
+        assertTrue(output.contains("Adds a new item to the inventory"));
+    }
+
+    @Test
+    public void testVerboseSpecificCommandHelp() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("add", true);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("add:"));
+        assertTrue(output.contains("Adds a new item to the inventory"));
+        assertTrue(output.contains("Usage:"));
+        assertTrue(output.contains("Example:"));
+    }
+
+    @Test
+    public void testUnknownCommand() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("unknown", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("Unknown command: unknown"));
+        assertTrue(output.contains("Available commands:"));
+    }
+
+    @Test
+    public void testHelpWithEmptyString() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("Available commands:"), "Should show general help for empty string input");
+    }
+
+    @Test
+    public void testHelpWithWhitespaceOnly() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("   ", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("Available commands:"), "Should show general help for whitespace-only input");
+    }
+
+    @Test
+    public void testHelpWithMixedCase() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("AdD", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("add:"), "Should recognize command regardless of case");
+    }
+
+    @Test
+    public void testHelpWithExtraArguments() throws PillException {
+        HelpCommand helpCommand = new HelpCommand("add extra args", false);
+        ItemMap itemMap = new ItemMap();
+        Storage storage = new Storage();
+
+        helpCommand.execute(itemMap, storage);
+
+        String output = outContent.toString().trim();
+        assertTrue(output.contains("add:"), "Should ignore extra arguments and show help for 'add'");
+    }
```

Comment:
Thanks for considering this case. Really helps with making our code more defensive.